### PR TITLE
Python3.5+

### PR DESCRIPTION
**Pull request by sfarrens**
- Dropped support for Python v2.7 and v3.4
- Removed all dependencies on `future`
- Removed all hacks for supporting Python v2.7
- Updated minimum dependency versions

> Note that this PR can only be merged after https://github.com/CEA-COSMIC/ModOpt/pull/57

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,29 +2,26 @@ sudo: required
...
+          dist: xenial
+          sudo: true
 
 before_install:
     - sudo apt-get update
     - sudo updatedb
-    - if [ $TRAVIS_PYTHON_VERSION == "2.7" ]; then
-        wget https://repo.continuum.io/miniconda/Miniconda2-latest-Linux-x86_64.sh -O miniconda.sh;
-      else
-        wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
```
Comment (AGrigis) [inline .travis.yml]: You need to install miniconda, you can't remove this line.
The Travis checks are failing for this reason.

**Comment by AGrigis** [comment]
To fix travis you need to install pyqt5. Indeed pyqtgraph does not support PySide2.
I think the best thing to do is to apply the MacOS hack for all platform:

MACOS_REQUIRES = [
    "pyqt5>=5.12.2"
]

File: .travis.yml (hunk 1 above)
Comment (sfarrens) [reply .travis.yml]: Fixed

**Comment by sfarrens** [comment]
@AGrigis It should be OK now.

**Pull request merged**